### PR TITLE
Release/2.0.0 beta.1

### LINKED ISSUE #292: Release 2.0.0-beta.1
We need to make an "interim" release that is not an official milestone. We are thinking to call it version 0.9. This indicates that a) it's not the truly official 1.0 release and b) 2.0 is not the appropriate number for an official release. 1.0 should be the official release since we are moving out of beta.

@gabrielhl @sambhav-gore @rbarilani What do you think?

### PR DESCRIPTION
**Pull request by gregdeane**
Closes #292 

@rbarilani I left the breaking changes info regarding `dc-dialog__overlay` in the changelog. Should we remove it or leave it?

A couple of tweaks had to be made to SC to make the new DC work correctly. Nothing major but some teams may be annoyed a bit.

### CODE REVIEW

**Comment by gabrielhl** [comment]
👍

**Comment by rbarilani** [comment]
👍 
@gregdeane I would leave the BC message.

File: dist/sass/_generated/_icon.scss [hunk 1]
```diff
@@ -243,23 +291,26 @@
 @mixin dc-icon-generated-selectors {
     @font-face {
         font-family: "DressCodeIcons";
```
Comment (rbarilani) [inline dist/sass/_generated/_icon.scss]: @gregdeane It seem that here the cache busting string ```?v=2.0.0-beta``` is not getting the right version

**Comment by gregdeane** [comment]
@rbarilani let us know if you need the release merge removed from `develop` branch. If I'm not available, please ask @cjhowald or @frontendherodk.

As for squashing commits... This may be possible currently but I believe once we upgrade to GitHub approval flow, it will no longer be possible. I might be wrong. That said, we may want to make our peace with unsquashed commits.

**Pull request merged**